cmd/podman/pull.go

```diff
@@ -124,7 +124,12 @@ func pullCmd(c *cli.Context) error {
 		}
 		imgID = newImage[0].ID()
 	} else {
-		newImage, err := runtime.ImageRuntime().New(getContext(), image, c.String("signature-policy"), c.String("authfile"), writer, &dockerRegistryOptions, image2.SigningOptions{}, true, forceSecure)
+		authfile := c.String("authfile")
+		envAuthfile := os.Getenv("REGISTRY_AUTH_FILE")
```

Comment:
Can probably move this inside the `if authfile = "" {` to avoid fetching the environment variable if we don't use it